addon/components/polaris-badge.js

```diff
@@ -65,44 +60,43 @@ export default class PolarisBadge extends Component {
    * @default null
    * @public
    */
-  progress = null;
+  progress;
 
   /**
    * Medium or small size. Use `small` only in the main navigation of an app frame..
    *
    * @type {String}
-   * @default 'medium'
+   * @default null
    * @public
    */
-  size = DEFAULT_SIZE;
+  @argWithDefault(DEFAULT_SIZE)
+  size;
+
+  /**
+   * Temporary workaround for not having useContext equivalents implemented yet.
+   * Pass `WithinFilterContext` true if badge is in a filter control component.
+   *
+   * TODO implement useContext or find Ember equivalent
+   */
+  @argWithDefault(false)
+  withinFilterContext;
 
   @notEmpty('progress')
   hasProgress;
 
   @notEmpty('status')
   hasStatus;
 
-  @computed('progress')
   get progressDescription() {
-    const { progress } = this;
-    if (isBlank(progress) || progress === 'default') {
-      return null;
-    }
-
-    return PROGRESS_LABELS[progress];
+    return this.args.progress ? PROGRESS_LABELS[this.args.progress] : null;
```

Comment:
Yeah, me and @tomnez have been chatting about this and we couldn't figure out where this `default` value was coming from, since the official Polaris implementation one doesn't really has anything like this, so we just agreed to drop it.

Reply:
Coolio. I have a suspicion that I added it in our initial implementation back when I was building components based on the HTML generated in the React examples rather than porting the React code itself 😅 Thanks for cleaning it up! 🙌